app/services/map_data_service.py

```diff
@@ -0,0 +1,288 @@
+            return None
+
+    def get_url(self, place_id: str) -> str:
+        try:
+            place = self.client.place(place_id=place_id)
+            return place.get("result", {}).get("url", "")
+        except Exception as e:
+            print(f"❌ Error: {e}")
+            return None
+
+    def process_place_data(self, place_data: Dict[str, Any]) -> Dict[str, Any]:
+        result = place_data.get("result", {})
+        
+        address_components = result.get("address_components", [])
+        address_extracted = {}
+        for component in address_components:
+            component_types = component.get("types", [])
+            long_name = component.get("long_name", "")
+            
+            for target_type in self.target_address_types:
+                if target_type in component_types:
+                    address_extracted[target_type] = long_name
+                    break
+        
+        processed_data = {
+            "postal_code": address_extracted.get("postal_code", ""),
+            "administrative_area_level_1": address_extracted.get("administrative_area_level_1", ""),
+            "administrative_area_level_2": address_extracted.get("administrative_area_level_2", ""),
+            "sublocality_level_1": address_extracted.get("sublocality_level_1", ""),
+            
+            "address": result.get("formatted_address", ""),
+            "phone_number": result.get("international_phone_number", ""),
+            "place_id": result.get("place_id", ""),
+            "url": result.get("url", ""),
+            
+            "coordinates": self.get_coordinates(result.get("geometry", {})),
+            
+            "rating": result.get("rating", 0.0),
+            "reviews": result.get("reviews", []),
+            
+            "price_level": result.get("price_level", 0),
+            "tags": result.get("types", []),
+            "wheelchair_accessible_entrance": result.get("wheelchair_accessible_entrance", False),
+            
+            "serves_beer": result.get("serves_beer", False),
+            "serves_breakfast": result.get("serves_breakfast", False),
+            "serves_brunch": result.get("serves_brunch", False),
+            "serves_dinner": result.get("serves_dinner", False),
+            "serves_lunch": result.get("serves_lunch", False),
+            "serves_vegetarian_food": result.get("serves_vegetarian_food", False),
+            "serves_wine": result.get("serves_wine", False),
+            "takeout": result.get("takeout", False),
+        }
+        
+        current_opening_hours = result.get("current_opening_hours")
+        if current_opening_hours:
+            processed_data["opening_hours"] = self.get_opening_hours(current_opening_hours)
+        
+        return processed_data
+    
+    def process_from_json(self, json_file_path: str) -> Dict[str, Any]:
+        with open(json_file_path, 'r', encoding='utf-8') as f:
+            place_data = json.load(f)
```

Comment:
No error handling if the file does not exist or is not valid JSON. Add try/except for robustness.